psydac/fem/splines.py

```diff
@@ -11,15 +11,37 @@
         find_span,
         basis_funs,
         collocation_matrix,
+        histopolation_matrix,
         breakpoints,
         greville,
-        elements_spans,
-        make_knots
+        make_knots,
+        elevate_knots,
+        basis_integrals,
         )
 from psydac.utilities.quadratures import gauss_legendre
+from psydac.utilities.utils import unroll_edges
 
 __all__ = ['SplineSpace']
 
+#===============================================================================
+def _periodic_intervals(domain, xgrid):
+
+    xA, xB = domain
+    assert all(xA <= xgrid)
+    assert all(xgrid <= xB)
+
+    if xgrid[0] == xA and xgrid[-1] == xB:
+        xleft  = xgrid[:-1]
+        xright = xgrid[1:]
+    elif xgrid[0] != xA:
+        xleft  = np.array([xgrid[-1] - (xB-xA), *xgrid[:-1]])
+        xright = xgrid
+    elif xgrid[-1] != xB:
+        xleft  = xgrid
+        xright = np.array([*xgrid[1:], xgrid[0] + (xB-xA)])
+
+    return xleft, xright
```

Comment:
Can you please check if this function is ever used in Psydac? (I think it's a leftover from a previous version of this file)

If it's not used, feel free to remove it.

Reply:
okay